src/common/snippets/src/op/subgraph.cpp

```diff
@@ -163,7 +163,11 @@ Shape snippets::op::Subgraph::canonicalize(const BlockedShapeVector& outputShape
             // todo: more complicated logics is needed if we want to merge smth else than blocked and planar
             // could be done by PartialShape::broadcast_merge_into, but this way is faster
             size_t startOffset = baseRank - inRank;
-            startOffset -= baseIsBlocked && baseOrder.back() != inOrder.back() ? 1 : 0;
```

Comment:
While writing the tests, I noticed that [1, 1, 2, 8] with [1, 4, 2, 8, 16] base is broadcasted to [1, 1, 2, 8, 1], which is fine. However, [1, 1, 2, 8] with [1, 8, 2, 8, 8] base (avx2 cpu for ex.) will be broadcasted to [1, 1, 1, 2, 8], and canonicalization will fail with "Failed to create broadcastable shapes in snippets canonicalization". So I added corresponding fix in this PR.